### PR TITLE
Add jest test template

### LINKED ISSUE #1: Add test templates too
I was enjoying using the [ava](https://www.npmjs.com/package/ava) test runner, so I'll mention that. It may not play nice with TypeScript, though.

I also would like to embrace a structure where we group test files next to their tested file in the `src` folder.

### PR DESCRIPTION
**Pull request by danfinlay**
Fixes #1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "module": "dist/index.es.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "yarn jest",
```
Comment (tmashuang) [inline package.json:9]:
```suggestion
    "test": "jest",
```

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+};
```
Comment (rekmarks) [inline jest.config.js:4]: We should add a newline here!

Comment (danfinlay) [reply jest.config.js:4]: `npx ts-jest config:init` disagrees, but I will add it :)

Comment (rekmarks) [reply jest.config.js]: 👑

File: package.json [hunk 3]
```diff
@@ -6,7 +6,8 @@
   "module": "dist/index.es.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "jest",
+    "test": "jest --watch",
```
Comment (tmashuang) [inline package.json:10]: Maybe this should be something like `test:watch`?
```suggestion
    "test:watch": "jest --watch",
```

Comment (rekmarks) [inline package.json:10]:
```suggestion
    "watch": "jest --watch",

```

**Review by rekmarks — DISMISSED** [review]
Just one typo/mis-paste!

File: README.md [hunk 4]
```diff
@@ -22,6 +22,12 @@ _Add examples here_
 
 _Add examples here_
 
+## Testing
+
+`npm test` or `yarn test` to run the tests once.
+
+To watch, install [jest](https://jestjs.io/) globally, and run `jest --watch`, or `npm run watch` or `yarn watch`.
```
Comment (rekmarks) [inline README.md:29]:
```suggestion
Run `yarn test` to run the tests once.

To run tests on file changes, run `yarn test:watch`.
```

**Review by rekmarks — DISMISSED** [review]

**Comment by whymarrh** [comment]
There is a fuller Jest config available [in MetaMask/safe-event-emitter](https://github.com/MetaMask/safe-event-emitter/blob/master/jest.config.js), one that has code coverage options set.

**Comment by rekmarks** [comment]
I added the Jest config suggested by @whymarrh.

**Review by rekmarks — APPROVED** [review]

**Pull request merged**